utils/build-script-impl

```diff
@@ -1391,37 +1392,7 @@ function swift_c_flags() {
 }
 
 function common_swift_flags() {
-    local host=$1
-    if [[ $(is_cross_tools_host ${host}) ]] ; then
-        echo -n "-target ${SWIFT_HOST_TRIPLE} -resource-dir $(build_directory ${host} swift)/lib/swift"
-        case $host in
-            android-aarch64)
-                echo -n " -sdk ${ANDROID_NDK}/platforms/android-${ANDROID_API_LEVEL}/arch-arm64"
-                echo -n " -tools-directory ${ANDROID_NDK}/toolchains/llvm/prebuilt/${LOCAL_HOST}/bin"
-                echo -n " -Xcc -I${ANDROID_NDK}/sysroot/usr/include"
-                echo -n " -Xcc -I${ANDROID_NDK}/sysroot/usr/include/${ANDROID_ARCH}-linux-android"
-                ;;
-            *)
-                echo "error: Swift cross-compilation flags for ${host} have not been set."
-                exit 1
-                ;;
-        esac
-    fi
-}
-
-function add_cross_cmake_options() {
-    local host=$1
-    local -n options=$2
-    case $host in
-        android-*)
-            options+=(
-                -DCMAKE_SYSTEM_NAME=Android
-                -DCMAKE_SYSTEM_VERSION=${ANDROID_API_LEVEL}
-                -DCMAKE_SYSTEM_PROCESSOR=${ANDROID_ARCH}
-                -DCMAKE_ANDROID_NDK:PATH=${ANDROID_NDK}
-            )
-            ;;
-    esac
+    echo -n "${SWIFT_FLAGS[@]} -module-cache-path \"${module_cache}\" "
```

Comment:
I can't remember if module_cache is expanded out or not. If it must be defined for this to work, please add an assert to make sure it is defined.

Reply:
It will always be defined, because [it is always set in the build loop for all products](https://github.com/apple/swift/blob/6ebddd84983cad27101de088dec2a4ad6b8b2add/utils/build-script-impl#L1669). I suppose now that I moved this to a function someone may later call it outside that build loop, say in the testing loop, though it is only called in the build loop now. If you're still worried about that, I can pass it in as a parameter to this function or add an assert.

Reply:
just the assert would be useful. That way if someone messes up we get a nice error msg.

Reply:
Done.